packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -1928,91 +1931,69 @@ export const getContributionBranches = async ({ state, effects }: Context) => {
 
 type RepositoriesActionOptions = {
   teamId: string;
-  bypassLoading?: boolean;
+  syncData?: boolean;
 };
 export const getRepositoriesByTeam = async (
-  { state, effects }: Context,
-  { teamId, bypassLoading = false }: RepositoriesActionOptions
+  { state, effects, actions }: Context,
+  { teamId, syncData = false }: RepositoriesActionOptions
 ) => {
   const { dashboard } = state;
-  const _repositories = dashboard.repositories;
-
-  // If we should show the loading state, we need to make two
-  // queries (synced and unsynced) since the waiting time to
-  // get up-to-date data from GitHub is too long.
-  if (!bypassLoading) {
-    try {
-      dashboard.repositories = { ..._repositories, [teamId]: null };
-
-      // First fetch data without syncing with GitHub
-      // to decrease waiting time.
-      const unsyncedRepositoriesData = await effects.gql.queries.getRepositoriesByTeam(
-        {
-          teamId,
-          syncData: false,
-        }
-      );
-      const unsyncedRepositories = unsyncedRepositoriesData?.me?.team?.projects;
-      if (!unsyncedRepositories) {
-        return;
-      }
 
-      dashboard.repositories = {
-        ..._repositories,
-        [teamId]: unsyncedRepositories.sort(sortByNameAscending),
-      };
-    } catch (error) {
-      effects.notificationToast.error(
-        'There was a problem getting your repositories'
-      );
-    }
-  }
-
-  // Fetch data synced with GitHub to make sure
-  // what we show is up-to-date.
   try {
-    const syncedRepositoriesData = await effects.gql.queries.getRepositoriesByTeam(
-      {
-        teamId,
-        syncData: true,
-      }
-    );
-    const syncedRepositories = syncedRepositoriesData?.me?.team?.projects;
-    if (!syncedRepositories) {
+    // First fetch data without syncing with GitHub
+    // to decrease waiting time.
+    const response = await effects.gql.queries.getRepositoriesByTeam({
+      teamId,
+      syncData,
+    });
+    const repositories = response?.me?.team?.projects;
+    if (!repositories) {
       return;
     }
 
-    dashboard.repositories = {
-      ..._repositories,
-      [teamId]: syncedRepositories.sort(sortByNameAscending),
+    dashboard.repositoriesByTeamId = {
+      ...dashboard.repositoriesByTeamId,
+      [teamId]: repositories.sort(sortByNameAscending),
     };
+
+    // If syncData was false, the initial call was made (faster), so we can
+    // safely trigger the second call to sync with GitHub (slower)
+    if (!syncData) {
+      actions.dashboard.getRepositoriesByTeam({ teamId, syncData: true });
```

Comment:
Hmm I get this code, but it feels strange that when `syncData` is `false`, we actually do sync data (but later). Sounds more like "lazy revalidation" instead of syncing? Can we clear up this concept by changing variable names in a way?

Reply:
I wondered if the two successive calls should be orchestrated from the UI instead of here, but unfortunately the UI action call is fire and continue, so I can't really do sequential work there. Maybe I could rename the param to `fetchCachedDataFirst`?

Reply:
Yeah something like that sounds good! I was thinking about `revalidateCache` or something like that, however, looking at the code again it does make a lot more sense to call it `fetchCachedDataFirst`. The function is a bit confusing at times because it calls itself. Don't have to change it, I might take another look at this function later to see what exactly is confusing me.